### PR TITLE
feat: edit contact details mutation

### PR DESCRIPTION
**Pull request by mritense**
Also adds default value and validation to edit fields. Contributes to https://ritense.tpondemand.com/entity/32565-contactgegevens-wijzigen

### CODE REVIEW

**Review by bddjong — COMMENTED** [review]
I think this implementation is OK for now. However, the code reads quite difficult and I wonder if messing with session storage to pass data between pages can be avoided. Lets keep this on our radar to improve.

File: packages/user-interface/src/components/detail-list/detail-list.tsx [hunk 1]
```diff
@@ -11,30 +11,61 @@ import {useMediaQuery} from '../../hooks';
...
+              {detail.loading ? (
+                <Skeleton width={isDesktop ? 200 : 150} />
+              ) : (
+                detail.value || EMPTY_VALUE
+              )}
             </span>
             {detail.showEditButton && (
               <Link
+                onClick={() =>
+                  setSessionStorageForProp(detail.translationKey, detail.value, detail.regex)
```
Comment (mjcarsjens) [inline packages/user-interface/src/components/detail-list/detail-list.tsx:65]: Not a big fan of this session storage setup, feels a bit hacky to me. 
Have you considered using the Context API for global state management for user information (e.g. data retrieved from graphql such as phone number and emailadress) and maybe creating for example a global validation file in which the regex expressions are stored based on certain identifiers (for example the same as used for the sessionStorage now) in a map and retrieved in the edit-account-page (in this case) based on the query string?

File: packages/user-interface/src/pages/account/account-page.tsx [hunk 2]
```diff
@@ -33,20 +32,21 @@ const AccountPage = () => {
...
-              headerTranslationKey: 'telefoonnummer',
+              translationKey: 'telefoonnummer',
               value:
-                contactLoading && !contactError
-                  ? ''
-                  : contactData?.getBurgerProfiel?.telefoonnummer || EMPTY_VALUE,
+                !contactLoading && !contactError && contactData?.getBurgerProfiel?.telefoonnummer,
               showEditButton: true,
+              loading: contactLoading,
+              regex: /^\d{10}$/,
```
Comment (mjcarsjens) [inline packages/user-interface/src/pages/account/account-page.tsx:49]: Is this also the server-side requirement for the phone field?
Feels very wrong to only allow 10-digit phone numbers without the possibility to provide a country code (e.g. +31 or 0031).

**Review by mjcarsjens — COMMENTED** [review]
@mritense left two comments, would be nice if you could take a look at them when you have time! For now this solution definitely works but I feel like it could use some optimalisation.

File: packages/user-interface/src/components/detail-list/detail-list.tsx (hunk 1 above)
Comment (mritense) [reply packages/user-interface/src/components/detail-list/detail-list.tsx:65]: I agree and this sounds like a good solution, I will try to implement what you suggest here

File: packages/user-interface/src/pages/account/account-page.tsx (hunk 2 above)
Comment (mritense) [reply packages/user-interface/src/pages/account/account-page.tsx:49]: This is the requirement I was handed down: 10 digits with no spaces. So I will keep it like this for now, but I agree it's an insufficient regex

**Pull request merged**